airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io/airbyte/cdk/read/cdc/CdcPartitionReader.kt

```diff
@@ -252,9 +235,8 @@ class CdcPartitionReader<T : Comparable<T>>(
             // At this point, if we haven't returned already, we need to close down the engine.
             log.info { "Shutting down Debezium engine: ${closeReason.message}." }
             // TODO : send close analytics message
-            // Launch engine.close() asynchronously but without an independent Job()
-            // This maintains the parent-child relationship, preventing premature socket closure
-            runBlocking { launch(Dispatchers.IO) { engine.close() } }
+            engineShuttingDown.set(true)
+            runBlocking { launch(Dispatchers.IO + Job()) { engine.close() } }
```

Comment:
`Job()` allows this line and the `EventConsumer.accept()` function to return immediately while engine close is happening on another thread. 
Leaving the function waiting may cause issues since accept is called by a debezium ahead, coming from the very engine we're trying to shut down.